Ports/qt6-qtbase/patches/0004-HACK-hard-pass-on-shared-memory-and-semaphores.patch

```diff
@@ -0,0 +1,110 @@
+From 7d6de8fce218809be7cc1531a2cca8d29a27083a Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Martin=20B=C5=99=C3=ADza?= <m@rtinbriza.cz>
+Date: Thu, 12 Aug 2021 19:16:44 +0200
+Subject: [PATCH 4/8] HACK: hard pass on shared memory and semaphores
+
+---
+ src/corelib/configure.cmake             | 4 ++--
+ src/corelib/configure.json              | 5 ++---
+ src/corelib/kernel/qcore_unix_p.h       | 2 +-
+ src/corelib/kernel/qsharedmemory_p.h    | 4 +++-
+ src/corelib/kernel/qsystemsemaphore_p.h | 3 +++
+ 5 files changed, 11 insertions(+), 7 deletions(-)
+
+diff --git a/src/corelib/configure.cmake b/src/corelib/configure.cmake
+index 1c585f5b..a96d8206 100644
+--- a/src/corelib/configure.cmake
++++ b/src/corelib/configure.cmake
+@@ -719,7 +719,7 @@ qt_feature("sharedmemory" PUBLIC
+     SECTION "Kernel"
+     LABEL "QSharedMemory"
+     PURPOSE "Provides access to a shared memory segment."
+-    CONDITION ( ANDROID OR WIN32 OR ( NOT VXWORKS AND ( TEST_ipc_sysv OR TEST_ipc_posix ) ) )
++    CONDITION ( QT_FEATURE_sharedmemory )
```

Comment:
Might be nicer in a "upstream this one day" kinda way to just add serenity to the list of opt-outers for these feature flags instead of just saying "false" or "circular dep pls". Side note, kinda surprising to see these if not INTEGRTY OR VXWORKS OR RTEMS in here. Who is compiling QT for realtime operating systems?

Reply:
Oh yeah, you're right. This is probably the nastiest thing I had to do because I for the life of me couldn't find a "right" way to do this without absolutely brutally hardcoding this in like 4 different places. Definitely a huge TODO on my list.

And I don't personally know what are those particular OSes used for but Qt is quite big in the embedded space (and automotive, especially).